### PR TITLE
Support for suppressing individual changes, to ease later overriding

### PR DESCRIPTION
**Pull request by deepfire**
This is a transparent change.

### CODE REVIEW

**Comment by deepfire** [comment]
cc @mpickering

**Comment by mpickering** [comment]
Can you give an example of how to use this and add it to the documentation?

**Comment by deepfire** [comment]
@mpickering, done!

File: scripts/generate-nix-overrides.hs [hunk 1]
```diff
@@ -85,9 +85,11 @@ main = do
                 [prefix, dir] -> return (prefix, dir)
                 _     -> fail "Usage: generate-nix-overrides [<prefix>, patchdir]"
   overrides <- generateOverrides prefix patchDir
-  putStrLn "{haskell}:"
-  putStrLn "let dontRevise = pkg: haskell.lib.overrideCabal pkg (old: { editedCabalFile = null; }); in"
-  putStrLn "let setCabalFile = pkg: file: haskell.lib.overrideCabal pkg (old: { postPatch = ''cp ${file} ${old.pname}.cabal''; }); in"
+  putStrLn "{haskell, suppressedPatches ? {}, suppressedCabals ? {}}:"
+  putStrLn "let dontRevise   = pkg: haskell.lib.overrideCabal pkg (old: { editedCabalFile = null; }); in"
+  putStrLn "let setCabalFile = pkg: file: if builtins.hasAttr ''${pkg.pname}'' suppressedCabals  then pkg else haskell.lib.overrideCabal pkg (old: { postPatch = ''cp ${file} ${old.pname}.cabal''; }); in"
+  putStrLn "let setPatch     = pkg: file: if builtins.hasAttr ''${pkg.pname}'' suppressedPatches then pkg else haskell.lib.appendPatch pkg file; in"
```
Comment (mpickering) [inline scripts/generate-nix-overrides.hs]: Not sure what the point of the value of the attribute is here, could you use a list for the same purpose?

Comment (deepfire) [reply scripts/generate-nix-overrides.hs]: @mpickering, just paranoid regarding worst-case performance, I guess..  I'll remake into a list if you think that's not a problem.

**Comment by phadej** [comment]
`head.hackage` upstream repository will be moving to GitLab  https://gitlab.haskell.org/ghc/head.hackage

Announced at https://mail.haskell.org/pipermail/ghc-devs/2019-July/017934.html

---

I'm closing this PR here, please reopen it on GitLab if it's still relevant.

**Pull request closed without merging**